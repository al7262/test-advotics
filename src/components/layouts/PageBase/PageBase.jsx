/* eslint-disable react/no-array-index-key */
/* eslint-disable react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { useLocation } from 'react-router';
import DashboardLogo from '../../../assets/dashboard.svg';
import Header from '../../elements/Header';
import styles from './index.css';

const navs = [
  { name: 'Dashboard', icon: DashboardLogo, to: '/' },
];

export default function PageBase({ children }) {
  return (
    <>
      <Header />
      <aside className={styles.aside}>
        <div className={styles.menu}>
          {[...Array(3).keys()].map((n) => <span key={n} />)}
        </div>
        <nav>
          {navs.map((data, n) => <NavSider data={data} key={n} />)}
        </nav>
      </aside>
      <main className={styles.main}>{children}</main>
    </>
  );
}

PageBase.defaultProps = {
  children: null,
};

PageBase.propTypes = {
  children: PropTypes.node,
};

export function NavSider({ data }) {
  const { pathname } = useLocation();
  const isActive = data.to === `/${pathname.split('/')[1].toLowerCase()}`;
  const className = [
    isActive && styles.active,
    styles['nav-item'],
  ].filter(Boolean).join(' ');

  return (
    <Link className={className} to={data.to}>
      <figure>
        <img src={data.icon} alt={`${data.name}-advotics-icon`} />
      </figure>
    </Link>
  );
}

NavSider.defaultProps = {
  data: {},
};

NavSider.propTypes = {
  data: PropTypes.object,
};
