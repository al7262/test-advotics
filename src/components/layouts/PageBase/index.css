.aside {
    position: fixed;
    width: 4.5em;
    height: 100vh;
    padding-top: 4em;
    background-color: var(--white-color);
}

.menu {
    width: 100%;
    padding: 1.5em;
}

.menu > span {
    height: .25em;
    width: 100%;
    border-radius: .5em;
    background-color: var(--black-color-4);
    margin-bottom: .375em;
    display: block;
}

.menu > span:last-child {
    width: 60%;
    margin-bottom: 0;
}

.nav-item {
    display: block;
    width: 100%;
    padding: .875em 1.5em;
}

.nav-item > figure {
    width: 1.5em;
    height: 1.5em;
    background-color: var(--white-color);
    border-radius: .25em;
    padding: .125em;
}

.nav-item > figure > image {
    width: 100%;
}

.nav-item.active {
    background-color: var(--black-color-4);
}

.main {
    padding: 5.75em 1em 1em 5.75em;
    height: 100vh;
}