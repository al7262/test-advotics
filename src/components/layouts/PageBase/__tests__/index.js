/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import PageBase from '../PageBase';

describe('src/components/layout/PageBase', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<PageBase />);
    expect(tree).toMatchSnapshot();
  });
});
