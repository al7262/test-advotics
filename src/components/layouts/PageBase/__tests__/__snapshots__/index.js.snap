// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/layout/PageBase render 1`] = `
<React.Fragment>
  <Header />
  <aside>
    <div>
      <span />
      <span />
      <span />
    </div>
    <nav>
      <NavSider
        data={
          Object {
            "icon": Object {
              "process": [Function],
            },
            "name": "Dashboard",
            "to": "/",
          }
        }
      />
    </nav>
  </aside>
  <main />
</React.Fragment>
`;
